### PR TITLE
roachtest: ensure all skipped tests are logged

### PR DESCRIPTION
**Pull request by herkolategan**
Previously, there was a bug where the last tests after the selected indexes were
not being logged as skipped. This change adds the length of the specs to the
list of selected indexes to ensure that the loop will skip all the tests between
the selected indexes, including the tail end of the tests.

Epic: None
Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/main_test.go [hunk 1]
```diff
@@ -483,3 +484,25 @@ func shuffleStrings(r *rand.Rand, strings []string) []string {
...
+
+	containsTestSkip := func(testName string) bool {
+		return strings.Contains(buf.String(), fmt.Sprintf("--- SKIP: %s", testName))
+	}
+
+	expectedSkippedTests := []string{"t1", "t2", "t3", "t5"}
+	for _, testName := range expectedSkippedTests {
+		assert.True(t, containsTestSkip(testName), "Expected skip message for %s not found in buffer", testName)
+	}
+	testName := "t4"
```
Comment (srosenberg) [inline pkg/cmd/roachtest/main_test.go:508]: Nit: maybe a comment to remind the reader that `t4` isn't skipped because of the seed.

**Review by srosenberg — APPROVED** [review]
Thanks!

**Review by golgeek — APPROVED** [review]

File: pkg/cmd/roachtest/main_test.go (hunk 1 above)
Comment (herkolategan) [reply pkg/cmd/roachtest/main_test.go:508]: Good idea, will add one.

File: pkg/cmd/roachtest/main.go [hunk 2]
```diff
@@ -488,18 +499,22 @@ func selectSpecs(
 	if atLeastOnePerPrefix {
 		sort.Ints(selectedIdxs)
 	}
+	// Add the last index+1 to the list of selected indexes. This is to ensure
+	// that the loop below skips all the tests between the selected indexes
+	selectedIdxs = append(selectedIdxs, len(specs))
```
Comment (DarrylWong) [inline pkg/cmd/roachtest/main.go]: nit: `selectedIdxs` is only used for printing skipped tests. Changing it to a map would let us get rid of the double for loop and not have to worry about this edge case. e.g.:

```
for i, s := range specs {
    if _, ok := selectedIdx[i]; !ok {
        print(s was skipped)
    }
}
```

Comment (herkolategan) [reply pkg/cmd/roachtest/main.go]: True, the implementation is a bit overly complicated for what it's doing. A set would make it much more readable and less error prone.

Comment (herkolategan) [reply pkg/cmd/roachtest/main.go]: I wonder if it's related to trying to keep the order (I mostly try not to break any of the original functionality when fixing a bug)? But then again we could just extract and sort the keys from a map. It's not like we're working with massive data structures here.

Comment (DarrylWong) [reply pkg/cmd/roachtest/main.go]: Ah I see, like trying to print all skipped `kv/*` tests together? That makes sense, but `selectedIdx` is derived by iterating through `specs`, so reiterating through it should give the same order?

**Review by DarrylWong — APPROVED** [review]

File: pkg/cmd/roachtest/main.go (hunk 2 above)
Comment (herkolategan) [reply pkg/cmd/roachtest/main.go]: Ah missed that part, yes it would.

**Comment by herkolategan** [comment]
TFTRs!

bors r=srosenberg,golgeek,DarrylWong
